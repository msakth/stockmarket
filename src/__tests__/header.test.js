import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Header from '../components/header';
import renderer from 'react-test-renderer';
import { Link, MemoryRouter } from 'react-router-dom';
configure({ adapter: new Adapter() });

describe('header component', () => {

    it('header renders without crashing', () => {
        shallow(<Header />);
    });

    it('header links match snapshot', () => {
      const headerComponent = renderer.create(
        <MemoryRouter>
           <div>
            <Link to="/" />
            <Link to="/tradelist" />
            <Link to="/tradeCalculations" />
          </div>
        </MemoryRouter>
      );

      const tree = headerComponent.toJSON();
      expect(tree).toMatchSnapshot();
    })
});
