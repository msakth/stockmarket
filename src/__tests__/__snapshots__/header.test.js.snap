// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component header links match snapshot 1`] = `
<div>
  <a
    href="/"
    onClick={[Function]}
  />
  <a
    href="/tradelist"
    onClick={[Function]}
  />
  <a
    href="/tradeCalculations"
    onClick={[Function]}
  />
</div>
`;
